middleware/formatRecipe.js

```diff
@@ -0,0 +1,148 @@
+            // Return the decimal number
+            return decimal;
+        } else{
+            return isNaN(amnt) ? undefined : parseInt(amnt)
+        }
+    }
+
+    const numericAmount = parseAmount(amount)
+
+    // Return the parsed ingredient object
+    return {
+        amount: numericAmount,
+        unit: numericAmount ? unit : undefined,
+        name: numericAmount ? name : ingredientString
+    }
+}
+
+const parseIngredientsArray = ingredients => ingredients.map(ing => parseIngredient(ing))
+
+// TODO: Consider checking if instructions are nested inside one big thing of text and pulling those options out in a future version
+const handleInstructions = instructions => {
+    // Check if we have type HowToStep for each element and change to just strings
+    const cleanInstructions = typeof instructions[0] === "object" && instructions[0] !== null ? 
+        instructions.map(el => el.text): 
+        instructions
+    
+    const instructionBlobRegex = /^\d+\.\s.*$/
+    if(cleanInstructions.length == 1 && instructionBlobRegex.test(cleanInstructions[0])){
+        const lines = cleanInstructions[0].split(/(?=\d+\.\s)/);
+        const splitInstructions = [];
+
+        lines.forEach(line => {
+            const match = line.match(/^\d+\.\s(.*)/);
+            if (match) {
+                splitInstructions.push(match[1].trim());
+            }
+        })
+
+        return splitInstructions;
+    } else{
+        return cleanInstructions
+    }
+}
+
+const handleYield = yield => {
+    if(Array.isArray(yield)){
+        yield = yield[0]
+    }
+
+    if(!isNaN(yield)){
+        yield += " servings"
+    }
+
+    return yield
+}
+
+const handleNutrition = nutrition => {
+    if(!nutrition){
+        return undefined
+    }
+
+    return {
+        servings: nutrition?.servingSize,
+        calories: nutrition?.calories,
+        fat: nutrition?.fatContent,
+        carbs: nutrition?.carbohydrateContent,
+        protein: nutrition?.proteinContent
+    }
+}
+
+const formatRecipe = (recipeData, recipeUrl) => {
+    const recipe = {
+        "title": recipeData?.name?.trim(),
+        "url": recipeUrl,
+        "imageUrl": Array.isArray(recipeData?.image) ? recipeData?.image[0]?.url : recipeData?.image?.url,
+        "prepTime": isoToMinutes(recipeData?.prepTime),
+        "cookTime": isoToMinutes(recipeData?.cookTime),
+        "totalTime": isoToMinutes(recipeData?.totalTime),
+        "nutrition": handleNutrition(recipeData?.nutrition),
+        "ingredients": parseIngredientsArray(recipeData?.recipeIngredient),
+        "instructions": handleInstructions(recipeData?.recipeInstructions),
+        "yield": handleYield(recipeData?.recipeYield),
+        "cuisine": Array.isArray(recipeData?.recipeCuisine) ? recipeData[0]?.recipeCuisine?.trim().toLowerCase() : recipeData?.recipeCuisine?.trim().toLowerCase(),
+        "tags": recipeData?.keywords?.split(",").map(keyword => keyword.trim().toLowerCase())
```

Comment:
this makes the most sense as strings that we can then make keywords for our search, etc.